marconi-chain-index/src/Marconi/ChainIndex/Node/Client/Retry.hs

```diff
@@ -33,6 +33,9 @@ data RetryState = RetryState
   , secondsBeforeNextRetry :: !Word64
   }
 
+-- TODO: this function is problematic because it is used both inside the
+-- ChainIndexT context and outside to initialize the config
+-- see Marconi.ChainIndex.Run.run
```

Comment:
Could you make the monad be a type parameter `m` and give it constraints `(MonadIO m, MonadUnliftIO m)`?

Or have two versions of the function, one of which wraps the other and lives in `ChainIndexT`?

Reply:
I'm not sure about using `MonadUnliftIO`. If I understand correctly, it looks like we'd be restricting our `ChainIndexT` monad stack to just having reader capabilites, and I don't have enough knowledge about the domain if we need to worry about this or not.

Reply:
Hmm, still, for the second solution I believe you end up explicitly implementing the `ChainIndexT -> IO` transformation. Unfortunately this isn't the only place in the code this transformation is needed _I think_, not 100% sure yet but wanted to add to my previous comment.